apps/ledger-live-mobile/src/contentCards/layouts/grid/index.tsx

```diff
@@ -2,39 +2,52 @@ import React from "react";
 import { useWindowDimensions } from "react-native";
 import { ContentLayoutBuilder } from "~/contentCards/layouts/utils";
 import { Flex } from "@ledgerhq/native-ui";
+import { WidthFactor } from "~/contentCards/layouts/types";
+import { useItemStyle as getItemStyle } from "./utils";
+import { ContentCardsType } from "~/dynamicContent/types";
 
 type Props = {
   styles?: {
     gap?: number;
+    widthFactor?: WidthFactor;
   };
 };
 
 const defaultStyles = {
-  gap: 6,
+  gap: 16,
+  widthFactor: WidthFactor.Half,
 };
 
 const Grid = ContentLayoutBuilder<Props>(({ items, styles: _styles = defaultStyles }) => {
+  const { width: windowWidth } = useWindowDimensions();
   const styles = {
     gap: _styles.gap ?? defaultStyles.gap,
+    widthFactor: _styles.widthFactor ?? defaultStyles.widthFactor,
   };
-
-  const width = useWindowDimensions().width / 2 - 20;
+  const cardWidth =
+    styles.widthFactor === WidthFactor.Full
+      ? windowWidth * styles.widthFactor - styles.gap * 2
+      : windowWidth * styles.widthFactor - styles.gap * 1.5;
+  const isStack =
+    styles.widthFactor === WidthFactor.Full && items[0].props.type === ContentCardsType.action;
 
   return (
     <Flex
       style={{
-        marginHorizontal: "auto",
-        justifyContent: "center",
-        width: "100%",
+        marginHorizontal: styles.gap,
+        justifyContent: "flex-start",
         flexDirection: "row",
         flexWrap: "wrap",
-        gap: styles.gap,
+        gap: isStack ? 0 : styles.gap,
       }}
     >
-      {items.map(item => {
+      {items.map((item, index) => {
         return (
-          <Flex key={item.props.metadata.id} style={{ width }}>
-            <item.component {...item.props} />
+          <Flex key={item.props.metadata.id} style={{ width: cardWidth }}>
+            <item.component
+              {...item.props}
+              itemStyle={getItemStyle(items.length === 1, index === 0, index === items.length - 1)}
```

Comment:
IMO I would pass as parameter only `index` and `Items.length`

And then in the function do 
`items.length === 1, index === 0, index === items.length - 1`

As they are calculated parameters it's better to handle it directly in the body function :)